Source/DiabloUI/settingsmenu.cpp

```diff
@@ -256,7 +256,7 @@ void UiSettingsMenu()
 
 		optionDescription[0] = '\0';
 
-		const char *titleText = shownMenu == ShownMenuType::Settings ? _("Settings") : selectedOption->GetName().data();
+		const char *titleText = shownMenu == ShownMenuType::Settings ? _("Settings").c_str() : selectedOption->GetName().data();
 		vecDialog.push_back(std::make_unique<UiArtText>(titleText, MakeSdlRect(PANEL_LEFT, UI_OFFSET_Y + 161, PANEL_WIDTH, 35), UiFlags::FontSize30 | UiFlags::ColorUiSilver | UiFlags::AlignCenter, 8));

```

Comment:
Won't that cause issues since `data()` isn't null terminated?

Reply:
Using `string_view::data()` is kinda dangerous, string_views are not guaranteed to be a null-terminated string so the length is pretty much required to read them safely. We get away with it in the current code because the view is over the whole of a `std::string`.

With the way UiArtText is constructed and used it's harder to make it take a string_view in the constructor, this is somewhat better than the current approach (can get rid `.c_str()` on the translate call) so still worthwhile imo

Reply:
Ok, I see that C+11 guaranties that `std::string` null terminated. Let hope `GetName()` is also derived from something null terminated :D